### PR TITLE
ICache: add label support & improve ICacheValueResolve#resolveAll doc

### PR DESCRIPTION
**Pull request by stephan-merkli**

### CODE REVIEW

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/cache/ICacheBuilder.java [hunk 1]
```diff
@@ -53,6 +54,15 @@ public interface ICacheBuilder<K, V> {
...
   ICacheBuilder<K, V> withCacheId(String cacheId);
 
+  /**
+   * Sets a label supplier.
+   *
+   * @param labelSupplier
+   *     Label supplier for label returned by {@link ICache#getLabel}.
+   * @return this builder
+   */
+  ICacheBuilder<K, V> withLabel(Supplier<String> labelSupplier);
```
Comment (andibur) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/cache/ICacheBuilder.java]:
```suggestion
  ICacheBuilder<K, V> withLabelSupplier(Supplier<String> labelSupplier);
```

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/cache/BasicCache.java [hunk 2]
```diff
@@ -37,13 +38,14 @@
...
   protected final String m_cacheId;
+  protected Supplier<String> m_labelSupplier;
   protected final ICacheValueResolver<K, V> m_resolver;
   protected final Map<K, V> m_cacheMap;
 
   protected final AbstractTransactionalMap<K, ?> m_transactionalMap; // is null if not transactional cache
 
-  public BasicCache(String cacheId, ICacheValueResolver<K, V> resolver, Map<K, V> cacheMap) {
-    this(cacheId, resolver, cacheMap, findTransactionalMap(cacheMap));
+  public BasicCache(String cacheId, Supplier<String> labelSupplier, ICacheValueResolver<K, V> resolver, Map<K, V> cacheMap) {
```
Comment (andibur) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/cache/BasicCache.java:56]: You could keep the old constructor and mark it deprecated. This way, current users do not necessarily have to adapt the code.

**Review by andibur — APPROVED** [review]
Thanks

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/cache/BasicCache.java [hunk 3]
```diff
@@ -37,13 +38,23 @@
 public class BasicCache<K, V> implements ICache<K, V> {
 
   protected final String m_cacheId;
+  protected Supplier<String> m_labelSupplier;
```
Comment (ralph-steiner) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/cache/BasicCache.java]: final Keyword

**Pull request merged**